### ISSUE TITLE
False-positive `Unused import`

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#10110]
## Environment

* **IntelliJ Rust plugin version:** 0.4.189.5197-223-nightly
* **Rust toolchain version:** 1.67.0 (fc594f156 2023-01-24) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2022.3.2 Ultimate Edition (IU-223.8617.56)
* **Operating system:** Linux 5.15.0-60-generic
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Steps to reproduce
```rust
#[proc_macro]
pub fn function_like_read_env_var(input: TokenStream) -> TokenStream {
    use std::fmt::Write;
    let v = std::env::var("FOO_ENV_VAR").unwrap();
    let mut s = String::new();
    write!(&mut s, "\"{}\"", v);
    return s.parse().unwrap();
}
```
